In [1]:
import requests
import json

In [4]:
sample_transaction = {
    "Time": 61290.0,
    "V1": 1.2288211502379,
    "V2": -0.0634077165201056,
    "V3": 0.274145142235826,
    "V4": 0.647465021810117,
    "V5": -0.0481345611508765,
    "V6": 0.372073028593297,
    "V7": -0.22423058741343,
    "V8": 0.0799390492455152,
    "V9": 0.640758817066441,
    "V10": -0.273053702248503,
    "V11": -1.25272793883718,
    "V12": 0.465078770741453,
    "V13": 0.400502115321077,
    "V14": -0.292841860600363,
    "V15": -0.10177401599731,
    "V16": -0.399835897844616,
    "V17": 0.0343356567914817,
    "V18": -0.783550254934187,
    "V19": 0.141344900433949,
    "V20": -0.0965659023514416,
    "V21": -0.129554448055005,
    "V22": -0.0837793282428063,
    "V23": -0.151661473916324,
    "V24": -0.700371597289218,
    "V25": 0.598550164523483,
    "V26": 0.491409070563651,
    "V27": 0.0029892597250263,
    "V28": 0.0017822861144491,
    "Amount": 11.5
}

In [8]:
response = requests.post(
    "http://127.0.0.1:5000/predict",
    json=sample_transaction
)

print("Status code:", response.status_code)
print(json.dumps(response.json(), indent=2))

Status code: 200
{
  "fraud_probability": 0.006913,
  "prediction": 0,
  "result": "LEGITIMATE",
  "threshold": 0.65
}


In [9]:
invalid_transaction = {
    "Time": 61290.0,
    "Amount": 11.5
}

response = requests.post(
    "http://127.0.0.1:5000/predict",
    json=invalid_transaction
)

print("Status code:", response.status_code)
print(json.dumps(response.json(), indent=2))

Status code: 400
{
  "error": "Missing required features",
  "missing_features": [
    "V1",
    "V2",
    "V3",
    "V4",
    "V5",
    "V6",
    "V7",
    "V8",
    "V9",
    "V10",
    "V11",
    "V12",
    "V13",
    "V14",
    "V15",
    "V16",
    "V17",
    "V18",
    "V19",
    "V20",
    "V21",
    "V22",
    "V23",
    "V24",
    "V25",
    "V26",
    "V27",
    "V28"
  ]
}


In [7]:
assert response.status_code == 200

result = response.json()

assert "fraud_probability" in result
assert "prediction" in result
assert "result" in result
assert "threshold" in result

print("API test passed successfully.")

API test passed successfully.


## API Validation Testing

The FraudLens API was tested using both invalid and valid transaction
requests.

### Invalid Request

A request containing only `Time` and `Amount` was submitted. The API
correctly returned HTTP 400 and identified the missing V1–V28 features.

### Valid Request

A complete transaction containing all 30 required features was submitted.
The API returned HTTP 200 and successfully generated a fraud probability of
0.006913.

Using the classification threshold of 0.65, the transaction was classified
as LEGITIMATE.

These tests confirm that the API performs both input validation and
successful model inference.